beacon-chain/core/blocks/withdrawals.go

```diff
@@ -76,3 +79,40 @@ func ProcessBLSToExecutionChange(st state.BeaconState, signed *ethpb.SignedBLSTo
 	err = st.UpdateValidatorAtIndex(message.ValidatorIndex, val)
 	return st, err
 }
+
+func ProcessWithdrawals(st state.BeaconState, withdrawals []*enginev1.Withdrawal) (state.BeaconState, error) {
+	expected, err := st.ExpectedWithdrawals()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get expected withdrawals")
+	}
+	if len(expected) != len(withdrawals) {
+		return nil, errors.New("invalid number of withdrawals")
+	}
+	for i, withdrawal := range withdrawals {
+		if withdrawal.WithdrawalIndex != expected[i].WithdrawalIndex {
+			return nil, errors.New("invalid withdrawal index")
```

Comment:
dont feel strongly about this, but perhaps we can turn these errors into variables. It might be useful later...
`var ErrInvalidWithdrawalIndex = errors.New("invalid withdrawal index")`